de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/wizard/ReqIfMappingPage.java

```diff
@@ -0,0 +1,232 @@
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.TableColumn;
+import org.eclipse.swt.widgets.TableItem;
+
+import de.dlr.sc.virsat.model.dvlm.Repository;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+
+/**
+ * A wizard page to review the requirement type which is used for imports
+ *
+ */
+public class ReqIfMappingPage extends WizardPage implements SelectionListener {
+
+	private static final String PAGE_TITLE = "ReqIF Mapping";
+	
+	private static final String COLUMN_SPECIFICATION_LABEL = "Specification";
+	private static final String COLUMN_CONTAINER_LABEL = "Container";
+	
+	private static final int COLUMN_NAME_WIDTH = 300;
+	private static final int COLUMN_TYPE_WIDTH = 300;
+
+
+	private static final int SPEC_COLUMN_INDEX = 1;
+
+	private Table table;
+	private List<TableItem> tableItems = new ArrayList<TableItem>();
+	private List<CCombo> editors = new ArrayList<CCombo>();
+	
+	private ReqIFImpl reqIfContent;
+
+	private Repository repository;
+	/**
+	 * Constructor
+	 */
+	protected ReqIfMappingPage() {
+		super(PAGE_TITLE);
+		setTitle(PAGE_TITLE);
+		setDescription("Check the specification elements to import and their container element in the system model.");
+	}
+
+
+	@Override
+	public void createControl(Composite parent) {
+
+		Composite content = new Composite(parent, SWT.NONE);
+		content.setLayout(new GridLayout());
+		content.setLayoutData(new GridData(GridData.FILL_BOTH | GridData.GRAB_HORIZONTAL | GridData.GRAB_HORIZONTAL));
+		setControl(content);
+
+		table = new Table(content, SWT.CHECK | SWT.BORDER | SWT.H_SCROLL);
+		table.setLayout(new GridLayout());
+		table.setLayoutData(new GridData(GridData.FILL_BOTH | GridData.GRAB_HORIZONTAL | GridData.GRAB_VERTICAL));
+		table.setHeaderVisible(true);
+		table.addSelectionListener(this);
+
+		TableColumn columnColumnName = new TableColumn(table, SWT.NULL);
+		columnColumnName.setWidth(COLUMN_NAME_WIDTH);
+		columnColumnName.setText(COLUMN_SPECIFICATION_LABEL);
+
+		TableColumn columnType = new TableColumn(table, SWT.NONE | SWT.DROP_DOWN);
+		columnType.setText(COLUMN_CONTAINER_LABEL);
+		columnType.setWidth(COLUMN_TYPE_WIDTH);
+
+	}
+
+	/**
+	 * Hand over reqIF content to this page
+	 * @param reqIFImpl the reqIF model
+	 * @param projectRepo the project's repository to get potential SEI containers from
+	 */
+	public void setInput(ReqIFImpl reqIFImpl, Repository projectRepo) {
+		this.reqIfContent = reqIFImpl;
+		this.repository = projectRepo;
+		cleanTable();
+		EList<Specification> specList = reqIFImpl.getCoreContent().getSpecifications();
+		for (Specification spec : specList) {
+			TableItem item;
+			int index = specList.indexOf(spec);
+			if (tableItems.size() <= index) {
+				item = new TableItem(table, SWT.NULL);
+				tableItems.add(item);
+			} else {
+				item = tableItems.get(index);
+			}

```

Comment:
Here we override the existing items. We miss the case if we select a file with many specifications, go back and select a file with less specifications:
![grafik](https://user-images.githubusercontent.com/56025440/111333534-14027b00-8673-11eb-90f0-7db8702b37ac.png)
->
![grafik](https://user-images.githubusercontent.com/56025440/111333616-24b2f100-8673-11eb-93a9-c84de539ae27.png)

So we also have to remove unused items here.

Reply:
Nice catch :+1

Reply:
I'm disposing all unused editors now.